src/main/java/com/teamwizardry/librarianlib/core/common/OreDictionaryRegistrar.kt

```diff
@@ -29,7 +33,39 @@ object OreDictionaryRegistrar {
     @JvmStatic
     fun registerOre(name: String, block: Block) = registerOre(name) { ItemStack(block) }
 
+    @JvmStatic
+    fun registerSmeltingItem(item: () -> Item, output: () -> ItemStack, xp: Float) {
+        furnaceRecipeItems[item] = output to xp
+    }
+
+    @JvmStatic
+    fun registerSmeltingItem(item: Item, output: () -> ItemStack, xp: Float) = registerSmeltingItem({ item }, output, xp)
+
+    @JvmStatic
+    fun registerSmeltingStack(item: () -> ItemStack, output: () -> ItemStack, xp: Float) {
+        furnaceRecipeStacks[item] = output to xp
+    }
+
+    @JvmStatic
+    fun registerSmeltingStack(item: ItemStack, output: () -> ItemStack, xp: Float) = registerSmeltingStack({ item }, output, xp)
+
+    @JvmStatic
+    fun registerSmeltingBlock(item: () -> Block, output: () -> ItemStack, xp: Float) {
+        furnaceRecipeBlocks[item] = output to xp
+    }
+
+    @JvmStatic
+    fun registerSmeltingBlock(item: Block, output: () -> ItemStack, xp: Float) = registerSmeltingBlock({ item }, output, xp)
+
     @JvmStatic
     @SubscribeEvent(priority = EventPriority.LOWEST)
-    fun init(e: RegistryEvent.Register<Item>) = toRegister.forEach { stack, key -> OreDictionary.registerOre(key, stack()) }
+    fun init(e: RegistryEvent.Register<Item>) {
+        toRegister.forEach { stack, key -> OreDictionary.registerOre(key, stack()) }
+        furnaceRecipeStacks.forEach { stack, out -> FurnaceRecipes.instance()
+                .addSmeltingRecipe(stack(), out.first(), out.second)}
+        furnaceRecipeItems.forEach { item, out -> FurnaceRecipes.instance()
+                .addSmelting(item(), out.first(), out.second)}
+        furnaceRecipeBlocks.forEach { block, out -> FurnaceRecipes.instance()
+                .addSmeltingRecipeForBlock(block(), out.first(), out.second)}
```

Comment:
Even though in 1.12.2 Furnace recipes aren't a registry yet, I still think it would be better to move these to the recipes registry event.

Reply:
Also, how about clearing the lists after usage? AFAIK we don't need to keep the lambda refs.